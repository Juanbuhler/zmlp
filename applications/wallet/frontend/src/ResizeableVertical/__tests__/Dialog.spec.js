import TestRenderer from 'react-test-renderer'

import ResizeableVerticalDialog from '../Dialog'

describe('<ResizeableVertical />', () => {
  it('should not render when larger than minExpandedSize', () => {
    const component = TestRenderer.create(
      <ResizeableVerticalDialog
        size={400}
        startingSize={0}
        minExpandedSize={300}
      />,
    )

    expect(component.toJSON()).toEqual(null)
  })

  it('should display collapse when dragging down', () => {
    const component = TestRenderer.create(
      <ResizeableVerticalDialog
        size={200}
        startingSize={300}
        minExpandedSize={300}
      />,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should display expand when dragging up', () => {
    const component = TestRenderer.create(
      <ResizeableVerticalDialog
        size={200}
        startingSize={0}
        minExpandedSize={300}
      />,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
