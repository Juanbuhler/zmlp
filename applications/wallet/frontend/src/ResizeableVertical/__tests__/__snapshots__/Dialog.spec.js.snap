// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResizeableVertical /> should display collapse when dragging down 1`] = `
.emotion-0 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 17px;
  font-height: 20px;
  font-weight: 500;
  overflow: hidden;
  color: #808080;
}

<div
  className="emotion-0"
>
  Release to 
  collapse
  .
</div>
`;

exports[`<ResizeableVertical /> should display expand when dragging up 1`] = `
.emotion-0 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 17px;
  font-height: 20px;
  font-weight: 500;
  overflow: hidden;
  color: #808080;
}

<div
  className="emotion-0"
>
  Release to 
  expand
  .
</div>
`;
